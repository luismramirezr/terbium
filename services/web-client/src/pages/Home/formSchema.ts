import * as yup from 'yup';

const schema = yup.object().shape({
  eml: yup.string().email('This must be a valid email').required('This is a required field'),
  psw: yup.string().required('This is a required field'),
});

export default schema;