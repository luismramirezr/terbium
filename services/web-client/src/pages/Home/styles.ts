import styled from 'styled-components';
import { darken } from 'polished';

export const Container = styled.div`
  height: 100%;
  width: 100%;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  color: #fff;

  div {
    max-width: 800px;
    width: 100%;
    flex: auto;
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;

    h1 {
      color: #ff79c6;
      padding: 10px 0;
    }

    h2 {
      padding: 10px 0;
    }

    p {
      line-height: 1.5rem;
      padding: 5px 0;
      text-align: center;
    }

    button {
      margin: 20px 0;
      height: 44px;
      padding: 0 25px;
      color: #fff;
      background: #3b9eff;
      font-weight: bold;
      border: 0;
      border-radius: 4px;
      transition: background 0.2s;

      &:hover {
        background: ${darken(0.03, '#3b9eff')};
      }
    }
  }
`;

export const Footer = styled.footer`
  width: 100%;
  display: flex;
  justify-content: center;
  align-items: center;
  padding: 50px 0;
  justify-self: flex-end;

  a {
    color: #fff;
    transition: opacity 0.2s;
    opacity: 0.8;

    &:hover {
      opacity: 1;
    }
  }
`;
