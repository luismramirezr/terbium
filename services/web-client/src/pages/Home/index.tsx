import React from 'react';

import { FaGithub } from 'react-icons/fa';
import { Container, Footer } from './styles';

export default function App() {

  return (
    <Container>
      <div>
        <h1>
          {'<'} Coding Labs {'/>'}
        </h1>
        <h2>Create-React-App Template</h2>
        <p>
          This template comes with a bunch of features to help you get started.
        </p>
        <p>
          Redux-Saga is already implemented with Reactotron for debugging and
          reduxsauce to reduce cluttering.
        </p>
        <p>
          There is a sample action to produce a notification with
          react-toastify.
        </p>
        <button type="button">
          Show me!
        </button>
        <p>For templating and styling, styled-components is used.</p>
        <p>
          Some other functions include: Prettier/ESLint with Airbnb guide,
          pre-commit hooks, extended support for Jest tests, among others .
        </p>
        <h2>Happy Coding!</h2>
      </div>
      <Footer>
        <a href="https://github.com/codinglabsdev">
          <FaGithub size={32} />
        </a>
      </Footer>
    </Container>
  );
}
