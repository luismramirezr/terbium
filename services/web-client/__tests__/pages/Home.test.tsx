import React from 'react';
import * as renderer from 'react-test-renderer';
import { mount } from 'enzyme';

import Home from '~/pages/Home';

describe('pages/Home', () => {
  const setupComponent = () => <Home />;

  it('renders', () => {
    const component = renderer.create(setupComponent());
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('has login inputs', () => {
    const component = mount(setupComponent());
    expect(component.find('TextField')).toExist();
    expect(component.find('PasswordField')).toExist();
    expect(component.find('Button').at(0).text()).toContain('Log In');
    expect(component.find('Button').at(1).text()).toContain('Forgot Password?');
  });
});
