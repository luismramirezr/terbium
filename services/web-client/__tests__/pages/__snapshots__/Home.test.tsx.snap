// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages/Home renders 1`] = `
<div
  className="sc-AxirZ jotJAS"
>
  <div>
    <div
      className="MuiContainer-root MuiContainer-maxWidthXs"
    >
      <div
        className="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded"
      >
        <div
          className="MuiBox-root MuiBox-root-1"
        >
          <form
            autoComplete="new-password"
            onSubmit={[Function]}
          >
            <div
              className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3 MuiGrid-direction-xs-column MuiGrid-align-items-xs-center"
            >
              <div
                className="MuiGrid-root MuiGrid-item"
              >
                <img
                  alt="Terbium"
                  src="terbium@4x.png"
                />
              </div>
              <div
                className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 MuiGrid-item MuiGrid-direction-xs-column"
              >
                <div
                  className="MuiGrid-root MuiGrid-item"
                >
                  <div
                    className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
                  >
                    <div
                      className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedStart MuiOutlinedInput-adornedStart MuiInputBase-marginDense MuiOutlinedInput-marginDense"
                      onClick={[Function]}
                    >
                      <div
                        className="MuiInputAdornment-root MuiInputAdornment-positionStart MuiInputAdornment-marginDense"
                      >
                        <svg
                          color="#3f51b5"
                          fill="currentColor"
                          height="1em"
                          stroke="currentColor"
                          strokeWidth="0"
                          style={
                            Object {
                              "color": "#3f51b5",
                            }
                          }
                          viewBox="0 0 1024 1024"
                          width="1em"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M928 160H96c-17.7 0-32 14.3-32 32v640c0 17.7 14.3 32 32 32h832c17.7 0 32-14.3 32-32V192c0-17.7-14.3-32-32-32zm-40 110.8V792H136V270.8l-27.6-21.5 39.3-50.5 42.8 33.3h643.1l42.8-33.3 39.3 50.5-27.7 21.5zM833.6 232L512 482 190.4 232l-42.8-33.3-39.3 50.5 27.6 21.5 341.6 265.6a55.99 55.99 0 0 0 68.7 0L888 270.8l27.6-21.5-39.3-50.5-42.7 33.2z"
                          />
                        </svg>
                      </div>
                      <input
                        aria-invalid={false}
                        autoFocus={false}
                        className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedStart MuiOutlinedInput-inputAdornedStart MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
                        disabled={false}
                        name="eml"
                        onAnimationStart={[Function]}
                        onBlur={[Function]}
                        onChange={[Function]}
                        onFocus={[Function]}
                        placeholder="Email"
                        required={false}
                        type="text"
                      />
                      <fieldset
                        aria-hidden={true}
                        className="PrivateNotchedOutline-root-2 MuiOutlinedInput-notchedOutline"
                        style={
                          Object {
                            "paddingLeft": 8,
                          }
                        }
                      >
                        <legend
                          className="PrivateNotchedOutline-legend-3"
                          style={
                            Object {
                              "width": 0.01,
                            }
                          }
                        >
                          <span
                            dangerouslySetInnerHTML={
                              Object {
                                "__html": "&#8203;",
                              }
                            }
                          />
                        </legend>
                      </fieldset>
                    </div>
                  </div>
                </div>
                <div
                  className="MuiGrid-root MuiGrid-item"
                >
                  <div
                    className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
                  >
                    <div
                      className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedStart MuiOutlinedInput-adornedStart MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd MuiInputBase-marginDense MuiOutlinedInput-marginDense"
                      onClick={[Function]}
                    >
                      <div
                        className="MuiInputAdornment-root MuiInputAdornment-positionStart MuiInputAdornment-marginDense"
                      >
                        <svg
                          color="#3f51b5"
                          fill="currentColor"
                          height="1em"
                          stroke="currentColor"
                          strokeWidth="0"
                          style={
                            Object {
                              "color": "#3f51b5",
                            }
                          }
                          viewBox="0 0 1024 1024"
                          width="1em"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M832 464h-68V240c0-70.7-57.3-128-128-128H388c-70.7 0-128 57.3-128 128v224h-68c-17.7 0-32 14.3-32 32v384c0 17.7 14.3 32 32 32h640c17.7 0 32-14.3 32-32V496c0-17.7-14.3-32-32-32zM332 240c0-30.9 25.1-56 56-56h248c30.9 0 56 25.1 56 56v224H332V240zm460 600H232V536h560v304zM484 701v53c0 4.4 3.6 8 8 8h40c4.4 0 8-3.6 8-8v-53a48.01 48.01 0 1 0-56 0z"
                          />
                        </svg>
                      </div>
                      <input
                        aria-invalid={false}
                        autoComplete="new-password"
                        autoFocus={false}
                        className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedStart MuiOutlinedInput-inputAdornedStart MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
                        disabled={false}
                        name="psw"
                        onAnimationStart={[Function]}
                        onBlur={[Function]}
                        onChange={[Function]}
                        onFocus={[Function]}
                        placeholder="Password"
                        required={false}
                        type="password"
                      />
                      <button
                        className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall"
                        disabled={false}
                        onBlur={[Function]}
                        onClick={[Function]}
                        onDragLeave={[Function]}
                        onFocus={[Function]}
                        onKeyDown={[Function]}
                        onKeyUp={[Function]}
                        onMouseDown={[Function]}
                        onMouseLeave={[Function]}
                        onMouseUp={[Function]}
                        onTouchEnd={[Function]}
                        onTouchMove={[Function]}
                        onTouchStart={[Function]}
                        tabIndex={0}
                        type="button"
                      >
                        <span
                          className="MuiIconButton-label"
                        >
                          <svg
                            color="#3f51b5"
                            fill="currentColor"
                            height="1em"
                            stroke="currentColor"
                            strokeWidth="0"
                            style={
                              Object {
                                "color": "#3f51b5",
                              }
                            }
                            viewBox="0 0 1024 1024"
                            width="1em"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <path
                              d="M942.2 486.2Q889.47 375.11 816.7 305l-50.88 50.88C807.31 395.53 843.45 447.4 874.7 512 791.5 684.2 673.4 766 512 766q-72.67 0-133.87-22.38L323 798.75Q408 838 512 838q288.3 0 430.2-300.3a60.29 60.29 0 0 0 0-51.5zm-63.57-320.64L836 122.88a8 8 0 0 0-11.32 0L715.31 232.2Q624.86 186 512 186q-288.3 0-430.2 300.3a60.3 60.3 0 0 0 0 51.5q56.69 119.4 136.5 191.41L112.48 835a8 8 0 0 0 0 11.31L155.17 889a8 8 0 0 0 11.31 0l712.15-712.12a8 8 0 0 0 0-11.32zM149.3 512C232.6 339.8 350.7 258 512 258c54.54 0 104.13 9.36 149.12 28.39l-70.3 70.3a176 176 0 0 0-238.13 238.13l-83.42 83.42C223.1 637.49 183.3 582.28 149.3 512zm246.7 0a112.11 112.11 0 0 1 146.2-106.69L401.31 546.2A112 112 0 0 1 396 512z"
                            />
                            <path
                              d="M508 624c-3.46 0-6.87-.16-10.25-.47l-52.82 52.82a176.09 176.09 0 0 0 227.42-227.42l-52.82 52.82c.31 3.38.47 6.79.47 10.25a111.94 111.94 0 0 1-112 112z"
                            />
                          </svg>
                        </span>
                      </button>
                      <fieldset
                        aria-hidden={true}
                        className="PrivateNotchedOutline-root-2 MuiOutlinedInput-notchedOutline"
                        style={
                          Object {
                            "paddingLeft": 8,
                          }
                        }
                      >
                        <legend
                          className="PrivateNotchedOutline-legend-3"
                          style={
                            Object {
                              "width": 0.01,
                            }
                          }
                        >
                          <span
                            dangerouslySetInnerHTML={
                              Object {
                                "__html": "&#8203;",
                              }
                            }
                          />
                        </legend>
                      </fieldset>
                    </div>
                  </div>
                </div>
                <div
                  className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-justify-xs-center"
                >
                  <label
                    className="MuiFormControlLabel-root"
                  >
                    <span
                      aria-disabled={false}
                      className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-6 MuiCheckbox-root MuiCheckbox-colorPrimary MuiIconButton-colorPrimary"
                      onBlur={[Function]}
                      onDragLeave={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      onKeyUp={[Function]}
                      onMouseDown={[Function]}
                      onMouseLeave={[Function]}
                      onMouseUp={[Function]}
                      onTouchEnd={[Function]}
                      onTouchMove={[Function]}
                      onTouchStart={[Function]}
                      tabIndex={null}
                    >
                      <span
                        className="MuiIconButton-label"
                      >
                        <input
                          className="PrivateSwitchBase-input-9"
                          data-indeterminate={false}
                          disabled={false}
                          name="rememberMe"
                          onChange={[Function]}
                          type="checkbox"
                        />
                        <svg
                          aria-hidden={true}
                          className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                          />
                        </svg>
                      </span>
                    </span>
                    <span
                      className="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
                    >
                      <p
                        className="MuiTypography-root MuiTypography-body2"
                      >
                        Remember me
                      </p>
                    </span>
                  </label>
                </div>
              </div>
              <div
                className="MuiGrid-root MuiGrid-item"
              >
                <button
                  className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
                  disabled={false}
                  onBlur={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex={0}
                  type="submit"
                >
                  <span
                    className="MuiButton-label"
                  >
                    Log In
                  </span>
                </button>
              </div>
              <div
                className="MuiGrid-root MuiGrid-item"
              >
                <a
                  aria-disabled={false}
                  className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textSizeSmall MuiButton-sizeSmall"
                  href="#forgot-password"
                  onBlur={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex={0}
                >
                  <span
                    className="MuiButton-label"
                  >
                    Forgot Password?
                  </span>
                </a>
              </div>
            </div>
          </form>
        </div>
      </div>
    </div>
  </div>
</div>
`;
